### PR TITLE
UI fixes for loading state, empty tree, added toast for error, fixed no indicies error

### PR DESCRIPTION
**Pull request by sumukhswamy**
### Description
[Describe what this change achieves]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/SQLPage/table_view.tsx [hunk 1]
```diff
@@ -448,7 +501,7 @@ export const TableView = ({ http, selectedItems, updateSQLQueries, refreshTree }
 
   return (
     <>
-      <EuiFlexGroup>
+      <>
```
Comment (ps48) [inline public/components/SQLPage/table_view.tsx:504]: Do we need double fragments?

File: public/components/SQLPage/table_view.tsx [hunk 2]
```diff
@@ -483,16 +536,17 @@ export const TableView = ({ http, selectedItems, updateSQLQueries, refreshTree }
           <EuiFlexGroup alignItems="center" direction="column">
             <EuiFlexItem grow={false}>
               <EuiEmptyPrompt
-                iconType="alert"
-                iconColor="danger"
+                icon={<EuiIcon type="database" size="m" />}
+                iconColor="subdued"
```
Comment (ps48) [inline public/components/SQLPage/table_view.tsx:540]: Should the banner change on error from the backend?

File: public/components/Main/main.tsx [hunk 3]
```diff
@@ -962,33 +963,38 @@ export class Main extends React.Component<MainProps, MainState> {
...
                   width: '400px',
-                  maxHeight: '1200px',
-                  overflowY: 'auto',
-                  overflowX: 'hidden',
+                  height: 'calc(100vh - 254px)',
                 }}
               >
-                <EuiFlexGroup direction="column">
-                  <EuiFlexItem>
+                <EuiFlexItem grow={false}>
```
Comment (ps48) [inline public/components/Main/main.tsx:969]: Is this flex item needed here? I don't see it part of any flex group

File: public/components/PPLPage/PPLPage.tsx [hunk 4]
```diff
@@ -147,17 +149,31 @@ export class PPLPage extends React.Component<PPLPageProps, PPLPageState> {
...
+                isDisabled={this.props.asyncLoading}
+              >
+                Explain
+              </EuiButton>
+            </EuiFlexItem>
+          ) : (
+            <EuiFlexItem
+              grow={false}
+              onClick={() =>
+                this.props.updatePPLQueries('source = <datasource>.<database>.<table> | head 10')
```
Comment (ps48) [inline public/components/PPLPage/PPLPage.tsx:167]: Should move the query to common/constants

File: common/utils/async_query_helpers.ts [hunk 5]
```diff
@@ -50,14 +50,15 @@ export const pollQueryStatus = (id: string, http: CoreStart['http'], callback) =
         callback({ status: status });
         setTimeout(() => pollQueryStatus(id, http, callback), POLL_INTERVAL_MS);
       } else if (status === 'failed') {
-        callback({ status: 'FAILED', results: [] });
+        const results = res.data.resp;
+        callback({ status: 'FAILED', error: results.error });
```
Comment (ps48) [inline common/utils/async_query_helpers.ts:54]: Have we tested the acceleration flyout with the function change here?

**Comment by ps48** [comment]
@paulstn @sumukhswamy Can you please update the tests?

File: public/components/SQLPage/table_view.tsx (hunk 2 above)
Comment (sumukhswamy) [reply public/components/SQLPage/table_view.tsx:540]: this is for the initial databases loading, when error comes there is a text saying Error fetching data as we discussed on call

File: public/components/SQLPage/table_view.tsx (hunk 1 above)
Comment (paulstn) [reply public/components/SQLPage/table_view.tsx:504]: removed the double fragments

File: public/components/Main/main.tsx (hunk 3 above)
Comment (paulstn) [reply public/components/Main/main.tsx:969]: Removed it

File: public/components/PPLPage/PPLPage.tsx (hunk 4 above)
Comment (paulstn) [reply public/components/PPLPage/PPLPage.tsx:167]: moved query to constants

**Review by ps48 — APPROVED** [review]

**Review by mengweieric — APPROVED** [review]

**Pull request merged**